packages/lib-auth/src/client.js

```diff
@@ -0,0 +1,318 @@
+   */
+  async getAccessToken() {
+    if (!this._state.accessToken) {
+      return ''
+    } else if (this._isAccessTokenValid()) {
+      return this._state.accessToken
+    } else {
+      return this._refreshTokenData()
+    }
+  }
+
+  /**
+   * Requests a cross-site request forgery (CSRF) token, which is used in the main authentication-related actions.
+   *
+   * @async
+   * @private
+   * @returns {Promise} Resolves to the CSRF token.
+   */
+  async _getCSRFToken () {
+    try {
+      const url = `/users/sign_in?now=${Date.now()}`
+      const config = { withCredentials: true }
+      const response = await this._httpClient.head(url, config)
+      const csrfToken = response.headers['x-csrf-token']
+      return csrfToken
+    } catch (error) {
+      console.error(error)
+      throw error
+    }
+  }
+
+  /**
+   * Tries to get the Panoptes session cookie from the browser's cookie jar.
+   *
+   * @private
+   * @returns {string} The JWT from the browser cookie, or an empty string
+   */
+  _getJWTFromBrowser () {
+    if (isNode) {
+      return ''
+    }
+    const { cookieName } = this._config
+    const cookies = cookie.parse(document.cookie)
+    return cookies[cookieName] || ''
+  }
+
+  /**
+   * Extracts the JWT from a response object's `set-cookie` header.
+   *
+   * @param {Object} response - The response object
+   * @param {Object} response.headers - The headers on the response object
+   * @param {Array|string} response.headers.set-cookie - The `set-cookie` header
+   */
+  _getJWTFromResponse (response) {
+    const setCookie = response.headers['set-cookie']
+    const cookies = (setCookie instanceof Array) ? setCookie : [setCookie]
+    return cookies.reduce((jwt, cookieString) => {
+      if (jwt) {
+        return jwt
+      }
+      const parsed = cookie.parse(cookieString)
+      return parsed[this._config.cookieName] || null
+    }, null)
+  }
+
+  /**
+   * Takes the new token data and merges it into the client's state
+   *
+   * @private
+   * @param {Object} tokenData - The token data returned from the API.
+   * @param {string} tokenData.access_token - The access token used to authenticate requests.
+   * @param {string} tokenData.created_at - The UNIX timestamp for the access token's creation.
+   * @param {string} tokenData.expires_in - The number of seconds before the access token expires.
+   * @param {string} tokenData.refresh_token - A token that can be exchanged for a new access token.
+   * @param {string} tokenData.scope - The scopes granted to the access token.
+   * @returns {void}
+   */
+  _handleNewTokenData (tokenData) {
+    this._state = {
+      accessToken: tokenData['access_token'],
+      accessTokenCreatedAt: tokenData['created_at'],
+      accessTokenExpiresAt: tokenData['created_at'] + tokenData['expires_in'],
+      accessTokenExpiresIn: tokenData['expires_in'],
+      refreshToken: tokenData['refresh_token'],
+      scope: tokenData.scope
+    }
```

Comment:
`client._state` is a public variable so this makes the refresh token visible via `client._state.refreshToken`. Your refresh token should be private just to you.